### PR TITLE
[pentest] Add ECC256 Keygen SCA test

### PR DESCRIPTION
**Pull request by nasahlpa**
This commit adds the ECC256 key generation side-channel penetration test to the codebase.

The host code is located in lowRISC/ot-sca#347

### CODE REVIEW

File: sw/device/tests/penetrationtests/firmware/ecc256_keygen_sca.c [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+uint32_t random_number[kEcc256CoordNumWords];
+
+uint32_t ecc256_fixed_number[kEcc256CoordNumWords];
+
+/**
+ * Adds two integers storred in byte arrays.
+ *
+ * Adds the integer stored in source array to the integer stored in
+ * destination aray.
+ * The user needs to ensure that dest_len isenough to store the result
```
Comment (m-temp) [inline sw/device/tests/penetrationtests/firmware/ecc256_keygen_sca.c:124]: Nit: typo: "isenough"

File: sw/device/tests/penetrationtests/firmware/ecc256_keygen_sca.c [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+ * without overflow.
+ *
+ * @param[in] dest  Location of the first input array and the result.
+ * @param[in] source  Location of the second input array.
+ * @param[in] dest_len   Length od the dest array in bytes.
```
Comment (m-temp) [inline sw/device/tests/penetrationtests/firmware/ecc256_keygen_sca.c:129]: Nit: typo: "od"

**Review by m-temp — COMMENTED** [review]
I had a look at the code.
LGTM, but I'm nor very familiar with writing device code.

File: sw/device/tests/penetrationtests/firmware/ecc256_keygen_sca.c (hunk 1 above)
Comment (nasahlpa) [reply sw/device/tests/penetrationtests/firmware/ecc256_keygen_sca.c]: Thanks, fixed.

File: sw/device/tests/penetrationtests/firmware/ecc256_keygen_sca.c (hunk 2 above)
Comment (nasahlpa) [reply sw/device/tests/penetrationtests/firmware/ecc256_keygen_sca.c]: Thanks, fixed.

**Review by wettermo — APPROVED** [review]
LGTM!

File: sw/device/tests/penetrationtests/firmware/otbn_sca.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  return OK_STATUS();
+}
+
+status_t handle_otbn_sca_init(ujson_t *uj) {
+  sca_init(kScaTriggerSourceOtbn, kScaPeripheralEntropy | kScaPeripheralIoDiv4 |
+                                      kScaPeripheralOtbn | kScaPeripheralCsrng |
+                                      kScaPeripheralEdn | kScaPeripheralHmac);
+
+  // Load p256 keygen from seed app into OTBN.
+  if (otbn_load_app(kOtbnAppP256KeyFromSeed).value != OTCRYPTO_OK.value) {
```
[the remark this answers is not in the document]
Comment (nasahlpa) [reply sw/device/tests/penetrationtests/firmware/otbn_sca.c:31]: Ah yeah, you are right. I've removed it :)

**Comment by nasahlpa** [comment]
Currently not building as after a rebase the image size exceeds the available IMEM. This issue is resolved by #22717

**Pull request closed without merging**